src/main/java/com/spring/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package com.spring.onlinebookstore.service.cart;
+
+import com.spring.onlinebookstore.dto.cart.CreateCartItemDto;
+import com.spring.onlinebookstore.dto.cart.UpdateCartItemDtoRequest;
+import com.spring.onlinebookstore.dto.cart.UpdateCartItemDtoResponse;
+import com.spring.onlinebookstore.dto.shoppingcart.ShoppingCartDto;
+import com.spring.onlinebookstore.exception.EntityNotFoundException;
+import com.spring.onlinebookstore.mapper.CartItemMapper;
+import com.spring.onlinebookstore.mapper.ShoppingCartMapper;
+import com.spring.onlinebookstore.model.Book;
+import com.spring.onlinebookstore.model.CartItem;
+import com.spring.onlinebookstore.model.ShoppingCart;
+import com.spring.onlinebookstore.model.User;
+import com.spring.onlinebookstore.repository.book.BookRepository;
+import com.spring.onlinebookstore.repository.cartitem.CartItemRepository;
+import com.spring.onlinebookstore.repository.shoppingcart.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByIdFetchUserAndCartItems(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by user id: " + userId
+                ));
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addItemToShoppingCart(CreateCartItemDto createCartItemDto, Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByIdFetchUserAndCartItems(userId)
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find shopping cart by user id: " + userId
+                ));
+        Book book = bookRepository.findById(createCartItemDto.bookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Can't find book by book id: " + createCartItemDto.bookId()
+                ));
+        CartItem cartItem = cartItemMapper.toModel(createCartItemDto);
```

Comment:
It's an example of implementation

```suggestion
        cart.getCartItems().stream()
                .filter(item -> item.getBook().getId().equals(itemDto.bookId()))
                .findFirst()
                .ifPresentOrElse(item -> item.setQuantity(item.getQuantity() + itemDto.quantity()),
                        () -> addCartItemToCart(itemDto, book, cart));
        repository.save(cart);
        return cartMapper.toDto(cart);
```